modules/shortcodes/dailymotion.php

```diff
@@ -93,40 +119,98 @@ function dailymotion_shortcode( $atts ) {
 		$params = shortcode_new_to_old_params( $atts );
 		parse_str( $params, $atts_new );
 
-		foreach( $atts_new as $k => $v ) {
+		foreach ( $atts_new as $k => $v ) {
 			$atts[ $k ] = $v;
 		}
 	}
 
-	if ( isset( $atts['id'] ) ) {
+	if ( isset( $atts['id'] ) && ! empty( $atts['id'] ) ) {
 		$id = $atts['id'];
 	} else {
 		return '<!--Dailymotion error: bad or missing ID-->';
 	}
 
-	if ( ! empty( $content_width ) ) {
-		$width = min( 425, intval( $content_width ) );
-	} else {
-		$width = 425;
+	/*set width and height using provided parameters if any */
+	$width  = isset( $atts['width'] )  ? intval( $atts['width'] )  : 0 ;
+	$height = isset( $atts['height'] ) ? intval( $atts['height'] ) : 0 ;
+
+	if ( ! $width && ! $height ) {
+		if ( ! empty( $content_width ) ) {
+			$width = absint( $content_width );
+		} else {
+			$width = 425;
+		}
+		$height = $width / 425 * 334;
+	} elseif ( ! $height ) {
+		$height = $width / 425 * 334;
+	} elseif ( ! $width ) {
+		$width = $height / 334 * 425;
+	}
+
+	// Video URL
+	$video_url = esc_url( 'https://www.dailymotion.com/embed/video/' . $atts['id'] );
+
+	/**
+	 * Let's add parameters if needed.
+	 *
+	 * @see https://developer.dailymotion.com/player
+	 */
+	$player_params = array();
+
+	if ( isset( $atts['autoplay'] ) && '1' === $atts['autoplay'] ) {
+		$player_params['autoplay'] = '1';
+	}
+	if ( isset( $atts['endscreen-enable'] ) && '0' === $atts['endscreen-enable'] ) {
+		$player_params['endscreen-enable'] = '0';
 	}
+	if ( isset( $atts['mute'] ) && '1' === $atts['mute'] ) {
+		$player_params['mute'] = '1';
+	}
+	if ( isset( $atts['sharing-enable'] ) && '0' === $atts['sharing-enable'] ) {
+		$player_params['sharing-enable'] = '0';
+	}
+	if ( isset( $atts['start'] ) && ! empty( $atts['start'] ) ) {
+		$player_params['start'] = abs( intval( $atts['start'] ) );
+	}
+	if ( isset( $atts['subtitles-default'] ) && ! empty( $atts['subtitles-default'] ) ) {
+		$player_params['subtitles-default'] = esc_attr( $atts['subtitles-default'] );
+	}
+	if ( isset( $atts['ui-highlight'] ) && ! empty( $atts['ui-highlight'] ) ) {
+		$player_params['ui-highlight'] = esc_attr( $atts['ui-highlight'] );
+	}
+	if ( isset( $atts['ui-logo'] ) && '0' === $atts['ui-logo'] ) {
+		$player_params['ui-logo'] = '0';
+	}
+	if ( isset( $atts['ui-start-screen-info'] ) && '0' === $atts['ui-start-screen-info'] ) {
+		$player_params['ui-start-screen-info'] = '0';
+	}
+	if ( isset( $atts['ui-theme'] ) && in_array( strtolower( $atts['ui-theme'] ), array( 'dark', 'light' ) ) ) {
+		$player_params['ui-theme'] = esc_attr( $atts['ui-theme'] );
+	}
+
+	// Add those parameters to the Video URL.
+	$video_url = add_query_arg(
+		$player_params,
+		$video_url
+	);
+
+	$id = urlencode( $id );
```

Comment:
@MathildeS maybe this one can be moved to this line in the file
```php
$id = $atts['id'];
```
when `$id` is finally defined.